pandas/io/sql.py

```diff
@@ -337,15 +337,24 @@ def read_sql(sql, con, index_col=None, coerce_float=True, params=None,
     """
     Read SQL query or database table into a DataFrame.
 
+    This function is a convenience wrapper around ``read_sql_table`` and
+    ``read_sql_query`` (for backward compatibility). It will delegate
+    to the specific function depending on the provided input.
+
+    For example, a SQL query will be routed to ``read_sql_query``, while a
```

Comment:
I would leave out "For example", as there are no other options.

(I would also keep those two paragraphs as a single paragraph, as it is explaining one topic.

Reply:
Done.